### PR TITLE
Update/add some models, Merge Map/Travel command script, Update all c…

### PR DESCRIPTION
**Pull request by nbely**
…ommands scripts to use Mongoose for database calls and follow new model structures.
Also have a lot of code in the main.js file for listening to reacts for the sign up sequence. I plan to port the majority of this to an external function soon in order to keep the main.js file a little cleanjer.

### CODE REVIEW

**Review by darocca — CHANGES_REQUESTED** [review]

File: main.js [hunk 1]
```diff
@@ -58,33 +60,112 @@ bot.on('message', message =>{
...
 
+// Set messageID for sign-up channel, and ID for trainer role to add
+const MessageNumber = '749970227898351648';
+const trainerRole = '749977507876438107';
+
+// Watch for reactions to sign-up channel
+bot.on('messageReactionAdd', async (reaction, user) => {
+
+	// function for starting profile creation if user reacts
+	const createProfile = async () => {
```
[the remark this answers is not in the document]
Comment (nbely) [reply main.js:89]: Sounds good, that was more or less my intention as well.

File: commands/files/newProfileQuestions.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
+[
```
[the remark this answers is not in the document]
Comment (nbely) [reply commands/files/newProfileQuestions.json:1]: This file is referenced in new.js under the commands folder. See lines 25, 37.

File: main.js [hunk 3]
```diff
@@ -58,33 +60,112 @@ bot.on('message', message =>{
...
 		message.channel.send(`>>> The command, ${command}, is not recognized. Please check your spelling or enter '-help' for a full list of commands.`);
+		console.log('Command not found');
 		break;
 	}
 
 });
 
 
+// Set messageID for sign-up channel, and ID for trainer role to add
+const MessageNumber = '749970227898351648';
```
[the remark this answers is not in the document]
Comment (nbely) [reply main.js:82]: Fixed in ti/map.

File: commands/map.js [hunk 4]
```diff
@@ -60,71 +59,65 @@ module.exports = {
...
 						const access = [];
 						const trcode = [];
+						const numEmoji = [];
+						const emojiCharacters = require('../scripts/emojiCharacters');
 
-						// onyl set valid to true if location was not found from entered arguments
+						// only set valid to true if location was not found from entered arguments
 						let valid = false;
-						if (typeof location != 'undefined') {
+						if (typeof location !== 'undefined' && location !== null) {
```
[the remark this answers is not in the document]
Comment (nbely) [reply commands/map.js:86]: Fixed in ti/map.

File: commands/map.js [hunk 5]
```diff
@@ -60,71 +59,65 @@ module.exports = {
...
 								else {
-									tempLoc = await findLocationbySimpNames(client, location.accessTo[i], '');
-									access[i] = tempLoc.loc;
-									trcode[i] = tempLoc.simpLoc;
+									trcode[i] = location.accessedBy[i]._id;
 								}
 							}
+
+							// Add flying to travel options if appplicable
+							if (location.usableHMs.indexOf('fly') != -1) {
```
[the remark this answers is not in the document]
Comment (nbely) [reply commands/map.js:110]: Resolved all similar issues in ti/map

File: commands/map.js [hunk 6]
```diff
@@ -60,71 +59,65 @@ module.exports = {
...
 							}
 
-							// find location from argument. If not found, search with alternate location name
-							location = await findLocationbySimpNames(client, args[0], tempArea);
-							if (typeof location == 'undefined') {
-								location = await findLocationbyAltNames(client, args[0], tempArea);
+							// find location from argument
+							location = await Location.findOne({locNames: args[0], areaName: tempArea })
+								.populate({path: 'accessedBy', select: '_id name locNames areaName island usableHMs'});
+							if (location === null) {
```
[the remark this answers is not in the document]
Comment (nbely) [reply commands/map.js:65]: Using truthiness here should be good. Also good know that "" is considered falsey too. Changed battle status check to use this as well, and corrected any other applicable use cases in ti/map.

**Review by darocca — APPROVED** [review]

**Pull request merged**